components/list/list-item-drag-handle.js

```diff
@@ -0,0 +1,204 @@
+	previousElement: 'previous-element'
+});
+
+class ListItemDragHandle extends LitElement {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean, reflect: true },
+			text: { type: String },
+			_keyboardActive: { type: Boolean },
+		};
+	}
+
+	static get styles() {
+		return [ buttonStyles, css`
+			:host {
+				align-items: center;
+				display: flex;
+				margin: 0.25rem;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-list-item-drag-handle-dragger-button,
+			.d2l-list-item-drag-handle-keyboard-button {
+				display: grid;
+				grid-auto-rows: 1fr 1fr;
+				margin: 0;
+				min-height: 1.8rem;
+				padding: 0;
+				width: 0.9rem;
+			}
+			/* Firefox includes a hidden border which messes up button dimensions */
+			button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-list-item-drag-handle-dragger-button {
+				background-color: unset;
+			}
+			button,
+			button[disabled]:hover,
+			button[disabled]:focus {
+				background-color: var(--d2l-color-gypsum);
+				color: var(--d2l-color-ferrite);
+			}
+			button:hover,
+			button:focus {
+				background-color: var(--d2l-color-mica);
+			}
+			button[disabled] {
+				cursor: default;
+				opacity: 0.5;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this._keyboardActive = false;
+		this.disabled = false;
+	}
+
+	focus() {
+		const node = getFirstFocusableDescendant(this);
+		if (node) node.focus();
+	}
+
+	render() {
+		return html`${this._keyboardActive && !this.disabled ? this._renderKeyboardDragging() : this._renderDragger()}`;
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('_keyboardActive')) {
+			this.focus();
+		}
+	}
+
+	_dispatchAction(action) {
+		if (!action) return;
+		this.dispatchEvent(new CustomEvent('d2l-list-item-drag-handle-action', {
+			detail: { action },
+			bubbles: false
+		}));
+	}
+
+	_dispatchActionDown() {
+		this._dispatchAction(dragActions.down);
+	}
+
+	_dispatchActionUp() {
+		this._dispatchAction(dragActions.up);
+	}
+
+	_onBlur() {
+		this._keyboardActive = false;
+		if (!this._tabbing) {
+			this._dispatchAction(dragActions.save);
+		}
+	}
+
+	_handleActiveKeyboard(e) {
+		if (!this._keyboardActive) {
+			return;
+		}
+		let action = null;
+		switch (e.key) {
+			case UP:
+				action = dragActions.up;
+				break;
+			case DOWN:
+				action = dragActions.down;
+				break;
+			case HOME:
+				action = dragActions.first;
+				break;
+			case END:
+				action = dragActions.last;
+				break;
+			case TAB:
+				action = e.shiftKey ? dragActions.previousElement : dragActions.nextElement;
+				break;
+			case ESC:
+				action = dragActions.cancel;
+				this.updateComplete.then(() => this._keyboardActive = false);
+				break;
+			case ENTER:
+			case SPACE:
+			case RIGHT:
+				action = dragActions.save;
+				this.updateComplete.then(() => this._keyboardActive = false);
+				break;
+			default:
+				return;
+		}
+		this._dispatchAction(action);
+		e.preventDefault();
+	}
+
+	_handleInactiveKeyboard(e) {
```

Comment:
would it be better name `_activateKeyboard` ?

Reply:
I like this name better for this method. But this current name follows the pattern for the ones above it which are nice. Is there a better name for the `_handleActiveKeyboard`?